### ISSUE TITLE
Include support for SSVC

### ISSUE THREAD

**Issue by justmurphy** [oasis-tcs/csaf#803]
We should include support for SSVC, as discussed in #462 and during July TC meeting.

**Comment by justmurphy** [comment]
Reasoning:

As referenced in the following blog post from former Executive Assistant Director for Cybersecurity, Eric Goldstein: ["Transforming the Vulnerability Management Landscape"](https://www.cisa.gov/news-events/news/transforming-vulnerability-management-landscape), CISA believes the integration of Stakeholder-Specific Vulnerability Categorization (SSVC) is crucial for advancing vulnerability management practices across organizations.

SSVC enables organizations to prioritize their remediation efforts effectively by assessing various attributes of vulnerabilities, including exploitation status and technical impact.

We have recently added support for SSVC to our IT advisories seen at CISA's public CSAF repository: [https://github.com/cisagov/CSAF/tree/develop/csaf_files/IT/white/2024](https://github.com/cisagov/CSAF/tree/develop/csaf_files/IT/white/2024)

**Comment by santosomar** [comment]
I completely agree. CSAF should provide support for SSVC. We should also eventually support EPSS.

**Comment by sei-vsarvepalli** [comment]
I support this effort as well and would like to see SSVC representation available in CSAF. By the way we also have an updated SSVC schema that addresses a number of concerns raised by analysts. The official SSVC schema that we would like to support is here:

https://certcc.github.io/SSVC/data/schema/v1/Decision_Point_Value_Selection-1-0-1.schema.json 

An example CVE record with this representation of an SVC evaluation is provided here

https://github.com/CVEProject/cve-schema/blob/1c08e97929e22b1983557fe6ea5a9573831d49db/schema/docs/full-record-advanced-example.json#L134C1-L156C13

**Comment by tschmidtb51** [comment]
@sei-vsarvepalli Is it possible to update the JSON Schema that to Draft 2020-12?

**Comment by ahouseholder** [comment]
Connecting some dots here: 

- CERTCC/SSVC#637 covers the specific request in https://github.com/oasis-tcs/csaf/issues/803#issuecomment-2436346492
- CERTCC/SSVC#640 is the umbrella issue for CSAF+SSVC integration

**Comment by santosomar** [comment]
A motion was moved by Omar to include the changes suggested in this pull request, during the CSAF TC monthly meeting on 2024-10-30. The motion was seconded by Michael. The motion passed.

**Comment by tschmidtb51** [comment]
Todos:
- [x] include in schema
- [x] add in referenced schema
- [x] add link in informative references
- [x] mention in design consideration principles
- [x] add to vulnerability section
- [ ] add to guidance on size
- [ ] check that link is correct
- [ ] add conversion rule
- [ ] add tests:
  - [ ] ìd` must be consistent with `cve` (if CVE) or part of `ids`
  - [ ] for known `namespaces`/`version`/`role`, values must match given schema
  - [ ] warn on unknown versions
  - [ ] info on old versions for known `namespace`/`role`
  - [ ] info for unknown `namespaces`/`role`

**Comment by tschmidtb51** [comment]
- [ ] update according to result of https://github.com/CERTCC/SSVC/issues/697

**Comment by tschmidtb51** [comment]
- [ ] Check outcome of private namespace discussion => may need to add a should test not to use

**Comment by sei-vsarvepalli** [comment]
Namespace specification has been updated in our schema - sorry we didn't notify. I have run the tests again in my repository https://github.com/sei-vsarvepalli/csaf/tree/ssvc with the latest SSVC JSON Schema. Welcome to add further test 

```
 ./csaf_2.1/test/csaf_schema/run_tests.sh
 ./csaf_2.1/test/csaf_schema/run_tests.sh csaf_vex
 ./csaf_2.1/test/provider_schema/run_tests.sh
 ./csaf_2.1/test/aggregator_schema/run_tests.sh
(venv) vijaymac:csaf vssarvepalli$ 
```

**Comment by tschmidtb51** [comment]
> Namespace specification has been updated in our schema - sorry we didn't notify. I have run the tests again in my repository https://github.com/sei-vsarvepalli/csaf/tree/ssvc with the latest SSVC JSON Schema. Welcome to add further test
> 
> ```
>  ./csaf_2.1/test/csaf_schema/run_tests.sh
>  ./csaf_2.1/test/csaf_schema/run_tests.sh csaf_vex
>  ./csaf_2.1/test/provider_schema/run_tests.sh
>  ./csaf_2.1/test/aggregator_schema/run_tests.sh
> (venv) vijaymac:csaf vssarvepalli$ 
> ```

I guess, the critical ones are in ` ./csaf_2.1/test/validator/run_tests.sh`.

**Comment by sei-vsarvepalli** [comment]
Yeh all those test also run fine

```
 ./csaf_2.1/test/validator/run_tests.sh mandatory
 ./csaf_2.1/test/validator/run_tests.sh optional
 ./csaf_2.1/test/validator/run_tests.sh informative
```

**Comment by santosomar** [comment]
Thank you for the update and support @sei-vsarvepalli !

**Comment by tschmidtb51** [comment]
> Namespace specification has been updated in our schema - sorry we didn't notify.

@sei-vsarvepalli No worries. However, it looks like the most current version (using the `"pattern": "^(?=.{3,25}$)(x_)?[a-z0-9]{3}([/.-]?[a-z0-9]+){0,22}$"` for the `namespace`) is not yet available to the website...
